snapshotter/volume_integ_test.go

```diff
@@ -97,6 +86,22 @@ func TestGuestVolumeFrom_Isolated(t *testing.T) {
 	require.NoErrorf(t, err, "Failed to create microVM[%s]", vmID)
 	defer fcClient.StopVM(ctx, &proto.StopVMRequest{VMID: vmID})
 
+	// Add a stargz image.
+	// The volume directories must be specified since the host's containerd doesn't know about the image.
+	remoteImage := volume.FromGuestImage(
```

Comment:
@kzys This is one slight change. `GuestVolumeImageProvider` handles pulling the remote image in order to extract the volume. Since the pull itself is hidden inside `VolumeSet.AddFrom`, we need the VM to be created before creating image provider so we can give it the vsock path to be used later when the pull happens. 

This means that the flow is now:
1) Create volume set
2) Add non-lazy volumes
3) Prepare drive mount
4) Launch VM
5) Add lazy volumes
6) Prepare lazy volumes in VM

Before we would have added lazy volumes right after adding non-lazy volumes. I think this still mostly makes sense, but I wanted to highlight that.